src/Regex-Core/RxMatcher.class.st

```diff
@@ -94,7 +94,10 @@ RxMatcher >> allocateMarker [
 { #category : 'testing' }
 RxMatcher >> atBeginningOfLine [
 
-	^self position = 0 or: [self lastChar = Cr]
+	^self lastChar
+		ifNotNil: [ :lc | lc = Cr or: [ lc = Lf ] ]
```

Comment:
It does not what the method says. In a CRLF line ending being on the CR will return true although there is a leading character at the same line.